Panic on attempt to initialise config

Hi @caarlos0 

I get Panic on attempt to initialise config in the root of my repository.

```
chglog config
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x100c6b730]

goroutine 1 [running]:
github.com/goreleaser/chglog/pkg/commands.setupConfigCmd.func2(0x140001a4600?, {0x1011f07e0?, 0x0?, 0x0?})
	github.com/goreleaser/chglog@v0.4.0/pkg/commands/config.go:27 +0x30
github.com/spf13/cobra.(*Command).execute(0x140001a4600, {0x1011f07e0, 0x0, 0x0})
	github.com/spf13/cobra@v1.6.1/command.go:896 +0x4dc
github.com/spf13/cobra.(*Command).ExecuteC(0x140001a4000)
	github.com/spf13/cobra@v1.6.1/command.go:1044 +0x35c
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.6.1/command.go:968
main.main()
	github.com/goreleaser/chglog@v0.4.0/cmd/chglog/main.go:51 +0x27c
```

## Environent
- go version go1.20 darwin/arm64
- chglog version: v0.4.0
- git version 2.37.1 (Apple Git-137.1)
- os version: macOS 13.2 



## Notes
The same issue with `version` command

```
chglog version
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104c34a10]

goroutine 1 [running]:
github.com/goreleaser/chglog/pkg/commands.setupVersionCmd.func1(0x1400021e300?, {0x1051b87e0?, 0x0?, 0x0?})
	github.com/goreleaser/chglog@v0.4.0/pkg/commands/version.go:16 +0x30
github.com/spf13/cobra.(*Command).execute(0x1400021e300, {0x1051b87e0, 0x0, 0x0})
	github.com/spf13/cobra@v1.6.1/command.go:896 +0x4dc
github.com/spf13/cobra.(*Command).ExecuteC(0x1400021e000)
	github.com/spf13/cobra@v1.6.1/command.go:1044 +0x35c
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.6.1/command.go:968
main.main()
	github.com/goreleaser/chglog@v0.4.0/cmd/chglog/main.go:51 +0x27c
```

--- a/cmd/chglog/main.go
+++ b/cmd/chglog/main.go
@@ -45,8 +45,13 @@ func main() {
 		return config.ReadInConfig()
 	}
 
-	cmds := commands.AllCommands(config)
-	cmdRoot.AddCommand(cmds...)
+	cmdRoot.AddCommand(
+		commands.AddCmd(config),
+		commands.InitCmd(config),
+		commands.VersionCmd(config),
+		commands.ConfigCmd(config),
+		commands.FormatCmd(config),
+	)
 
 	if err := cmdRoot.Execute(); err != nil {
 		// nolint: gomnd, gocritic

--- a/pkg/commands/add.go
+++ b/pkg/commands/add.go
@@ -15,7 +15,7 @@ import (
 )
 
 // nolint: gocognit, funlen, gocritic
-func setupAddCmd(config *viper.Viper) (cmd *cobra.Command) {
+func AddCmd(config *viper.Viper) (cmd *cobra.Command) {
 	var (
 		input,
 		output,

--- a/pkg/commands/commands.go
+++ b/pkg/commands/commands.go
@@ -7,19 +7,6 @@ import (
 	"github.com/spf13/viper"
 )
 
-// AllCommands returns all commands supported.
-func AllCommands(config *viper.Viper) (cmds []*cobra.Command) {
-	cmds = append(cmds,
-		setupVersionCmd(config),
-		setupConfigCmd(config),
-		setupAddCmd(config),
-		setupInitCmd(config),
-		setupFormatCmd(config),
-	)
-
-	return cmds
-}
-
 func commonFlags(cmd *cobra.Command, config *viper.Viper) (*cobra.Command, *viper.Viper) {
 	var (
 		urgency, owner      string

--- a/pkg/commands/config.go
+++ b/pkg/commands/config.go
@@ -5,7 +5,7 @@ import (
 	"github.com/spf13/viper"
 )
 
-func setupConfigCmd(config *viper.Viper) (cmd *cobra.Command) {
+func ConfigCmd(config *viper.Viper) (cmd *cobra.Command) {
 	var pkg string
 	cmd = &cobra.Command{
 		Use:   "config",

--- a/pkg/commands/format.go
+++ b/pkg/commands/format.go
@@ -16,7 +16,7 @@ import (
 var ErrTemplateFlags = errors.New("--template and --template-file are mutually exclusive")
 
 // nolint: funlen, gocritic
-func setupFormatCmd(config *viper.Viper) (cmd *cobra.Command) {
+func FormatCmd(config *viper.Viper) (cmd *cobra.Command) {
 	var input,
 		output,
 		pkg,

--- a/pkg/commands/init.go
+++ b/pkg/commands/init.go
@@ -15,7 +15,7 @@ import (
 // ErrNoTags happens when a repository has no tags.
 var ErrNoTags = errors.New("no versioned releases found, check the output of `git tag`")
 
-func setupInitCmd(config *viper.Viper) (cmd *cobra.Command) {
+func InitCmd(config *viper.Viper) (cmd *cobra.Command) {
 	var output string
 	cmd = &cobra.Command{
 		Use:   "init [PATH]",
@@ -30,10 +30,6 @@ func setupInitCmd(config *viper.Viper) (cmd *cobra.Command) {
 		"o",
 		"changelog.yml",
 		"file to save the new changelog to")
-
-	cmd.PersistentPreRun = func(c *cobra.Command, args []string) {
-		cmd.Parent().PersistentPreRun(c, args)
-	}
 
 	cmd.RunE = func(c *cobra.Command, args []string) (err error) {
 		var (

--- a/pkg/commands/version.go
+++ b/pkg/commands/version.go
@@ -7,7 +7,7 @@ import (
 	"github.com/spf13/viper"
 )
 
-func setupVersionCmd(config *viper.Viper) (cmd *cobra.Command) {
+func VersionCmd(config *viper.Viper) (cmd *cobra.Command) {
 	cmd = &cobra.Command{
 		Use:   "version",
 		Short: "display version info",

